python/lsst/pipe/tasks/computeExposureSummary.py

```diff
@@ -0,0 +1,198 @@
+    def run(self, exposure, sources, background):
+        """Measure exposure statistics from the exposure, sources, and background.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.ExposureF`
+        sources : `lsst.afw.table.SourceCatalog`
+        background : `lsst.afw.math.BackgroundList`
+
+        Returns
+        -------
+        summary : `lsst.afw.image.ExposureSummary`
+        """
+        self.log.info("Measuring exposure statistics")
+
+        bbox = exposure.getBBox()
+
+        psf = exposure.getPsf()
+        if psf is not None:
+            shape = psf.computeShape(bbox.getCenter())
+            psfSigma = shape.getDeterminantRadius()
+            psfIxx = shape.getIxx()
+            psfIyy = shape.getIyy()
+            psfIxy = shape.getIxy()
+            im = psf.computeKernelImage(bbox.getCenter())
+            # The calculation of effective psf area is taken from
+            # meas_base/src/PsfFlux.cc#L112. See
+            # https://github.com/lsst/meas_base/blob/
+            # 750bffe6620e565bda731add1509507f5c40c8bb/src/PsfFlux.cc#L112
+            psfArea = np.sum(im.array)/np.sum(im.array**2.)
+        else:
+            psfSigma = np.nan
+            psfIxx = np.nan
+            psfIyy = np.nan
+            psfIxy = np.nan
+            psfArea = np.nan
+
+        wcs = exposure.getWcs()
+        if wcs is not None:
+            sph_pts = wcs.pixelToSky(lsst.geom.Box2D(bbox).getCorners())
+            raCorners = [float(sph.getRa().asDegrees()) for sph in sph_pts]
+            decCorners = [float(sph.getDec().asDegrees()) for sph in sph_pts]
+
+            sph_pt = wcs.pixelToSky(bbox.getCenter())
+            ra = sph_pt.getRa().asDegrees()
+            decl = sph_pt.getDec().asDegrees()
+        else:
+            raCorners = [float(np.nan)]*4
+            decCorners = [float(np.nan)]*4
+            ra = np.nan
+            decl = np.nan
+
+        photoCalib = exposure.getPhotoCalib()
+        if photoCalib is not None:
+            zeroPoint = 2.5*np.log10(photoCalib.getInstFluxAtZeroMagnitude())
+        else:
+            zeroPoint = np.nan
+
+        visitInfo = exposure.getInfo().getVisitInfo()
+        date = visitInfo.getDate()
+
+        if date.isValid():
+            observatory = visitInfo.getObservatory()
+            loc = EarthLocation(lat=observatory.getLatitude().asDegrees()*units.deg,
+                                lon=observatory.getLongitude().asDegrees()*units.deg,
+                                height=observatory.getElevation()*units.m)
+            obstime = Time(visitInfo.getDate().get(system=DateTime.MJD),
+                           location=loc, format='mjd')
+            coord = SkyCoord(ra*units.degree, decl*units.degree, obstime=obstime, location=loc)
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore')
+                altaz = coord.transform_to(AltAz)
+
+            zenithDistance = altaz.alt.degree
```

Comment:
Given that this is already known to ObservationInfo (and possibly directly calculated from the FITS headers)  it seems a bit wasteful to calculate it here from the ra/dec/time. Can we add zenith distance to visitInfo (like it is already in the butler exposure table)?

Reply:
So the fits header records boresight info, but we want this per-detector (which can vary significantly for the LSST FOV).  So "visitInfo" isn't the right place because it's not per-visit information.

Reply:
Ah. Can you add a comment to that effect then?